### PR TITLE
Add user names table and some basic admin commands

### PR DESCRIPTION
**Pull request by JesseWilliamson**
This should be fully compatible with the existing database but it's probably a good idea to make a backup copy of timelord.db first anyway. Two new slash commands (/leaderboard and /allusersums) and two new event subscriptions (team_join and user_change) are needed.

### CODE REVIEW

**Comment by ysims** [comment]
It would be useful to be able to choose a date, and all the data from that date to present date is used. Doesn't need to be in this PR though, can make a new one for that.

File: README.md [hunk 1]
```diff
@@ -18,4 +18,18 @@ Add the following environment variables to the `.env` file
...
 
 ## Usage
-Run app.py to launch Timelord
\ No newline at end of file
+Run app.py to launch Timelord
+
+### Slash Commands
+
+#### User Commands:
+- `/timelog` Opens a time logging form
```
Comment (LachlanCourt) [inline README.md:27]: I think there's `/help` as well isn't there?

File: app.py [hunk 2]
```diff
@@ -18,138 +18,148 @@
...
+        # Open an SQL connection and add entry to database containing user input
+        sqlc = database.SQLConnection()
+        sqlc.insert_timelog_entry(user_id, selected_date, minutes)
+
+        respond(f"Time logged: {time_input[0]} hours and {time_input[1]} minutes for date {selected_date}.")
+
+    except Exception as e:
+        # Show the user an error if they input anything other than two integers seperated by some character / characters
+        logger.exception("Invalid user input, failed to create time log entry")
+        respond("*Invalid input!* Please try again!")
```
Comment (LachlanCourt) [inline app.py]: It might be helpful to provide an example of the expected input as part of this error response

File: app.py [hunk 3]
```diff
@@ -172,24 +209,66 @@ def log_database(ack, body, respond, command, logger):
     else:
         respond("You must be an admin to use this command!")
 
+# Get a leaderboard with the top 10 contributors and their hours logged
+@app.command("/leaderboard")
+def leaderboard(ack, body, respond, logger, command):
+    ack()
+    try:
+        user_id = body['user_id']
+        num_users = int(command['text']) if command['text'] != "" else 10 # Defaults to 5 entries
```
Comment (LachlanCourt) [inline app.py]: I think either the comment or the code is incorrect here, is it 5 or 10?

**Review by LachlanCourt — COMMENTED** [review]
Just a few notes but otherwise looking good :)

**Review by LachlanCourt — APPROVED** [review]
Nice, I'll chuck the new commands up on slack and deploy it on friday :)

**Pull request merged**